nautobot/core/testing/filters.py

```diff
@@ -155,6 +165,116 @@ def test_tags_filter(self):
             qs_result = self.queryset.filter(tags=tags[0]).filter(tags=tags[1]).distinct()
             self.assertQuerysetEqualAndNotEmpty(filterset_result, qs_result)
 
+        def test_q_filter_exists(self):
+            """Test the `q` filter exists on a filterset, does not validate the filter works as expected."""
+            self.assertTrue(self.filterset.declared_filters.get("q"))
+
+        def _assert_valid_filter_predicates(self, obj, field_name):
+            self.assertTrue(
+                hasattr(obj, field_name),
+                f"`{field_name}` is an Invalid `q` filter predicate for `{self.filterset.__name__}`",
+            )
+
+        def _assert_nested_q_filter(self, obj, model_field_name, filter_field_name, lookup_method):
+            """
+            Asserts the validity of a nested filter in the queryset.
+
+            Args:
+                obj: The object on which the filter is applied.
+                model_field_name: The name of the field in the model.
+                filter_field_name: The name of the field being filtered.
+                lookup_method: The method used for the lookup.
+            """
+            filter_field_name, nested_field = filter_field_name.split("__", 1)
+            field = obj._meta.get_field(filter_field_name)
+
+            if isinstance(field, (ManyToOneRel, ManyToManyRel, ManyToManyField)):
+                rel_obj = getattr(obj, filter_field_name).first()
+            else:
+                rel_obj = getattr(obj, filter_field_name)
+
+            if "__" not in nested_field:
+                self._assert_non_nested_q_filter(rel_obj, model_field_name, nested_field, lookup_method)
+            else:
+                self._assert_nested_q_filter(rel_obj, model_field_name, nested_field, lookup_method)
```

Comment:
Maybe consider doing this with a `while` loop instead of recursion? Not blocking.